scripts/llnl_scripts/llnl_lc_build_tools.py

```diff
@@ -442,68 +457,144 @@ def set_axom_group_and_perms(directory):
-    write_build_info(pjoin(prefix,"info.json"), job_name)
+    write_build_info(pjoin(prefix, "info.json"), job_name)
+
+    repo_dir = get_repo_dir()
+    # Clean previously generated host-configs into TPL install directory
+    print "[Cleaning previously generated host-configs if they exist]"
+    host_configs = get_host_configs_for_current_machine(repo_dir, True)
+    for host_config in host_configs:
+        os.remove(host_config)
+
     # use uberenv to install for all specs
+    tpl_build_failed = False
     for spec in specs:
         start_time = time.time()
-        res = uberenv_install_tpls(prefix,spec,mirror_dir)
+        fullspec = "{0}+devtools".format(spec)
+        res = uberenv_build(prefix, fullspec, project_file, config_dir, mirror_dir)
         end_time = time.time()
         print "[build time: {0}]".format(convertSecondsToReadableTime(end_time - start_time))
         if res != 0:
             print "[ERROR: Failed build of tpls for spec %s]\n" % spec
-            # set perms, then early exit
-            # set proper perms for installed tpls
-            set_axom_group_and_perms(prefix)
-            # set proper perms for the mirror files
-            set_axom_group_and_perms(mirror_dir)
-            return res
+            tpl_build_failed = True
+            break
         else:
             print "[SUCCESS: Finished build tpls for spec %s]\n" % spec
-    # build the axom against the new tpls
-    res = build_and_test_host_configs(prefix, job_name, timestamp)
-    if res != 0:
-        print "[ERROR: build and test of axom vs tpls test failed.]\n"
-    else:
-        print "[SUCCESS: build and test of axom vs tpls test passed.]\n"
+
+    # Copy generated host-configs into TPL install directory
+    print "[Copying spack generated host-configs to TPL build directory]"
+    host_configs = get_host_configs_for_current_machine(repo_dir, True)
+    for host_config in host_configs:
+        dst = pjoin(prefix, os.path.basename(host_config))
+        if os.path.exists(host_config) and not os.path.exists(dst):
+            shutil.copy2(host_config, dst)
+
+    if not tpl_build_failed:
+        # build the axom against the new tpls
+        res = build_and_test_host_configs(prefix, job_name, timestamp, True)
+        if res != 0:
+            print "[ERROR: build and test of axom vs tpls test failed.]\n"
+        else:
+            print "[SUCCESS: build and test of axom vs tpls test passed.]\n"
+ 
     # set proper perms for installed tpls
     set_axom_group_and_perms(prefix)
     # set proper perms for the mirror files
     set_axom_group_and_perms(mirror_dir)
     return res
 
 
-def get_host_configs_for_current_machine(src_dir):
+def build_devtools(builds_dir, job_name, timestamp):
+    sys_type = get_system_type()
+    config_dir = "scripts/uberenv/spack_configs/{0}/devtools".format(sys_type)
+    project_file = "scripts/uberenv/devtools.json"
+
+    if "toss_3" in sys_type:
+        compiler_spec = "%gcc@8.1.0"
```

Comment:
Minor: Should this be in our ``specs.yaml``, and grabbed from there?

Reply:
It is very static.  it could be pulled into a new specs.yaml (maybe devtoolsspecs.yaml) then some logic to normalize the directory like spack does.  But I would prefer to leave that for another PR. Mainly because I'm tired of doing this atm.